src/carrot_cli/rest/templates.py

```diff
@@ -82,3 +140,43 @@ def subscribe(template_id, email):
 def unsubscribe(template_id, email):
     """Submits a request to CARROT's templates unsubscribe mapping"""
     return request_handler.unsubscribe("templates", template_id, email)
+
+
+def __process_wdl(params, files, wdl, type):
+    # If wdl is an http or gs uri, we'll add it to params
+    if wdl.startswith("http://") or wdl.startswith("https://") or wdl.startswith("gs://"):
+        params.append((f"{type}_wdl", wdl))
+    # Otherwise, assume wdl is a file, so we'll try to open it and put it in a files list
+    else:
+        try:
+            wdl_file = open(wdl, "rt")
+            files[f'{type}_wdl_file'] = (f'{type}.wdl', wdl_file)
```

Comment:
Use the with construct here so the file is automatically closed:

```suggestion
            with open(wdl, 'rt') as wdl_file:
                files[f'{type}_wdl_file'] = (f'{type}.wdl', wdl_file)
```

Reply:
This won't work and I'm curious for your opinion on an alternative approach to this problem.  The purpose of this function is to add the appropriate value for the wdl to either the params list or the files dict (depending on whether it's a http or gs uri, or it's a local file), formatted for the request_handler to be able to use it (which is dictated by the python requests api).  The files dict has to actually contain the file object, so if we use `with`, it'll close the file before it makes it to the request_handler.  I have some ideas for ways to deal with this, so I'll figure out which way I want to do it and then push those changes.

Reply:
Just pushed a commit for this

Reply:
AH.  That's weird.  I think your solution is fine.  The `finally` block effectively does the same thing.